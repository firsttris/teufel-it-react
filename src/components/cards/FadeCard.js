import React, {Component} from 'react';
import {FormattedMessage} from 'react-intl';
import PropTypes from 'prop-types';

class FilpCard extends Component {
    constructor(props) {
        super(props);
        this.state = {
            flipped: false
        };
    }

    onMouseLeave() {
        this.setState({flipped: false});
    }

    onMouseEnter() {
        this.setState({flipped: true});
    }

    onClick() {
        this.setState({flipped: true});
    }

    render() {
        const verticalCenterStyle = {
            display: 'inline-flex',
            alignItems: 'center'
        };
        const colorTransitionStyle = {transition: 'color 2s linear 1s'};
        colorTransitionStyle.color = this.state.flipped ? 'white' : '#f00';
        return (
            <div className="py-2 col-xs-12 col-md-6 col-lg-6"
                 onMouseLeave={this.onMouseLeave.bind(this)}
                 onMouseEnter={this.onMouseEnter.bind(this)}
                 onClick={this.onClick.bind(this)}>
                <div style={verticalCenterStyle}>
                    <i className={'fa fa-2x ' + this.props.icon} style={colorTransitionStyle} aria-hidden="true"/>
                    <span className="text-uppercase" style={{marginLeft: '10px', fontSize: '20px'}}>
                        <FormattedMessage id={this.props.title}/>
                    </span>
                </div>
                <p>
                    <FormattedMessage id={this.props.text}/>
                </p>
            </div>
        );
    }
}

FilpCard.propTypes = {
    icon: PropTypes.string,
    title: PropTypes.string,
    text: PropTypes.string
};

export default FilpCard;