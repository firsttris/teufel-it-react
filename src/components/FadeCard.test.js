import React from 'react';
import FadeCard from './FadeCard';
import renderer from 'react-test-renderer';

test('FadeCard changes the class when hovered', () => {
  const component = renderer.create(
    <FadeCard key="1" class="col-xs-12 col-md-6 col-lg-6" icon="fa-github" title="Hello" text="test"></FadeCard>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseEnter();
  // re-rendering
  tree = component.toJSON();

  expect(tree).toMatchSnapshot();
  // manually trigger the callback
  tree.props.onMouseLeave();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});