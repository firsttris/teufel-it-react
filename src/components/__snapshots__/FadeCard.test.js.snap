// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FadeCard changes the class when hovered 1`] = `
<div
  className="py-2 col-xs-12 col-md-6 col-lg-6"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "inline-flex",
      }
    }
  >
    <i
      aria-hidden="true"
      className="fa fa-2x fa-github"
      style={
        Object {
          "color": "#f00",
          "transition": "color 2s linear 1s",
        }
      }
    />
    <span
      className="text-uppercase"
      style={
        Object {
          "fontSize": "20px",
          "marginLeft": "10px",
        }
      }
    >
      Hello
    </span>
  </div>
  <p>
    test
  </p>
</div>
`;

exports[`FadeCard changes the class when hovered 2`] = `
<div
  className="py-2 col-xs-12 col-md-6 col-lg-6"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "inline-flex",
      }
    }
  >
    <i
      aria-hidden="true"
      className="fa fa-2x fa-github"
      style={
        Object {
          "color": "white",
          "transition": "color 2s linear 1s",
        }
      }
    />
    <span
      className="text-uppercase"
      style={
        Object {
          "fontSize": "20px",
          "marginLeft": "10px",
        }
      }
    >
      Hello
    </span>
  </div>
  <p>
    test
  </p>
</div>
`;

exports[`FadeCard changes the class when hovered 3`] = `
<div
  className="py-2 col-xs-12 col-md-6 col-lg-6"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "inline-flex",
      }
    }
  >
    <i
      aria-hidden="true"
      className="fa fa-2x fa-github"
      style={
        Object {
          "color": "#f00",
          "transition": "color 2s linear 1s",
        }
      }
    />
    <span
      className="text-uppercase"
      style={
        Object {
          "fontSize": "20px",
          "marginLeft": "10px",
        }
      }
    >
      Hello
    </span>
  </div>
  <p>
    test
  </p>
</div>
`;
